tests/ui/pattern/never_patterns.rs

```diff
@@ -0,0 +1,85 @@
+#![feature(never_patterns)]
+#![allow(incomplete_features)]
+
+enum Void {}
+
+fn main() {}
+
+// The classic use for empty types.
+fn safe_unwrap_result<T>(res: Result<T, Void>) {
+    let Ok(_x) = res;
+    let (Ok(_x) | Err(!)) = &res;
+    //~^ ERROR: is not bound in all patterns
+    let (Ok(_x) | Err(&!)) = res.as_ref();
+    //~^ ERROR: is not bound in all patterns
+}
+
+// Check we only accept `!` where we want.
+fn never_pattern_location() {
+    // Don't accept on a non-empty type.
+    match Some(0) {
+        None => {}
+        Some(!) => {}
+    }
```

Comment:
This says "don't accept", but I don't see an `//~ ERROR` here. Should this be an error, since `Some(_)` is not covered?

Reply:
If these are currently not operating the way they should, then maybe they should be split out into a new test file, and you can proactively file a bug + make it `// known-bug: #...`

Reply:
hm indeed this isn't implemented yet. I could say "shouldn't accept"?

Reply:
Well either do what I said above, or mark them as FIXME(never_patterns) in the test file, just so we are aware of it.